### PR TITLE
[WIP] Cleanup HWND related interop

### PR DESCRIPTION
**Pull request by hughbe**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/1508)

### CODE REVIEW

**Comment by dreddy-work** [comment]
Thanks @hughbe for this PR. We need to thoroughly validate these changes to make sure we are not regressing anything. Change pattern is straight but number of changes are huge. For core 3.0, We do not have app.config support for HDPI and validation of these changes need extra efforts.  We can take this PR for 3.1.
One thing i noticed in the changes is, OS validation/api availability check missing. was that intentional? if yes, please add some explanation. 
@RussKie FYI.

File: src/Common/src/Interop/User32/Interop.DpiAwarenessContext.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        public enum DpiAwarenessContext
+        {
+            DPI_AWARENESS_CONTEXT_UNSPECIFIED = 0,
```
[the remark this answers is not in the document]
Comment (hughbe) [reply src/Common/src/Interop/User32/Interop.DpiAwarenessContext.cs]: I believe that interop guidelines are to keep constants (i.e. enum values) as similar to their Windows name as possible

File: src/Common/src/Interop/User32/Interop.GetWindowPlacement.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        [DllImport(Libraries.User32, ExactSpelling = true, EntryPoint = "GetWindowPlacement")]
+        private static extern int GetWindowPlacementInternal(IntPtr hWnd, ref WINDOWPLACEMENT placement);
```
[the remark this answers is not in the document]
Comment (hughbe) [reply src/Common/src/Interop/User32/Interop.GetWindowPlacement.cs]: I'm following the pattern in `RltGetVersion`:
```cs
        [DllImport(Libraries.NtDll, EntryPoint="RtlGetVersion", ExactSpelling=true)]
        private static extern int RtlGetVersionInternal(ref RTL_OSVERSIONINFOEX lpVersionInformation);

        internal static unsafe int RtlGetVersion(out RTL_OSVERSIONINFOEX versionInfo)
        {
            versionInfo = new RTL_OSVERSIONINFOEX
            {
                dwOSVersionInfoSize = (uint)sizeof(RTL_OSVERSIONINFOEX)
            };
            return RtlGetVersionInternal(ref versionInfo);
        }
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/ComponentModel/Design/DesignerHostTests.cs [hunk 3]
```diff
@@ -1825,7 +1825,7 @@ public void DesignerHost_Add_RootDesignerDisposeThrowsDuringUnloadingDispose_Thr
             Assert.False(host.Loading);
         }
 
-        [Fact]
+        [Fact(Skip = "Unstable test, see: https://github.com/dotnet/winforms/issues/1460")]
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/ComponentModel/Design/DesignerHostTests.cs]: Please send this change (any similar in future) as separate PRs, as we will fast track them

File: src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.cs [hunk 4]
```diff
@@ -74,8 +74,8 @@ public class CheckedListBox : ListBox
 
         static CheckedListBox()
         {
-            LBC_GETCHECKSTATE = SafeNativeMethods.RegisterWindowMessage("LBC_GETCHECKSTATE");
-            LBC_SETCHECKSTATE = SafeNativeMethods.RegisterWindowMessage("LBC_SETCHECKSTATE");
+            LBC_GETCHECKSTATE = Interop.User32.RegisterWindowMessageW("LBC_GETCHECKSTATE");
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.cs]: @JeremyKuhne previously we had safe and unsafe native methods segregated, with all the refactoring and cleanup work (incl. this one) we're loosing this distinction.
Is this something we need to be concerned with?

Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.cs]: > Is this something we need to be concerned with?

No. It isn't a pattern that makes sense with CAS gone. https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1060-move-p-invokes-to-nativemethods-class?view=vs-2019

**Comment by hughbe** [comment]
Closing this temporarily.

**Pull request closed without merging**